hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -433,10 +433,33 @@ spec = parallel $ do
 
                 waitUntil [n1, n2] $ DecommitFinalized{headId = testHeadId, decommitTxId = txId decommitTx1}
 
-                send n1 (Decommit{decommitTx = decommitTx2})
+                send n2 (Decommit{decommitTx = decommitTx2})
                 waitUntil [n1, n2] $ DecommitApproved{headId = testHeadId, decommitTxId = txId decommitTx2, utxoToDecommit = utxoRefs [22]}
                 waitUntil [n1, n2] $ DecommitFinalized{headId = testHeadId, decommitTxId = txId decommitTx2}
 
+      it "can process transactions while decommit pending" $
+        shouldRunInSim $ do
+          -- NOTE: The simulated network has a block time of 20 (simulated) seconds.
+          withSimulatedChainAndNetwork $ \chain ->
+            withHydraNode aliceSk [bob] chain $ \n1 ->
+              withHydraNode bobSk [alice] chain $ \n2 -> do
+                openHead chain n1 n2
+
+                let decommitTx = SimpleTx 1 (utxoRef 1) (utxoRef 42)
+                send n2 (Decommit{decommitTx})
+                waitUntil [n1, n2] $
+                  DecommitRequested{headId = testHeadId, decommitTx, utxoToDecommit = utxoRefs [42]}
+                waitUntil [n1, n2] $
+                  DecommitApproved{headId = testHeadId, decommitTxId = 1, utxoToDecommit = utxoRefs [42]}
+
+                let normalTx = SimpleTx 2 (utxoRef 2) (utxoRef 3)
+                send n2 (NewTx normalTx)
+                waitUntilMatch [n1, n2] $ \case
+                  SnapshotConfirmed{snapshot = Snapshot{confirmed}} -> 2 `elem` confirmed
```

Comment:
would be great to assert the version hasn't been bumped here,
but it gets increased upon receiving DecommitFinalized

Reply:
`DecommitFinalized` doesn't contain the version information. How would we compare the versions?

Reply:
don't we have a mean to know the state of the head?

do you think is out of scope to add the version info to `DecommitFinalized`?

Reply:
We only have means of getting the Head UTxO afaik and adding the version to `DecommitFinalized` just to be able to assert something in test is not really adding much value to the users/head-operators, it's just something that could help us make sense but then I would add it to all of the messages if we deem it important.